safe/common/parameters/default_select_parameter_widget.py

```diff
@@ -151,3 +152,29 @@ def toggle_custom_value(self):
             self.custom_value.setDisabled(False)
         else:
             self.custom_value.setDisabled(True)
+
+    def toggle_input(self):
+        """Change behaviour of radio button based on input."""
+        # If input is a field, then disable all radio button.
+        if self.input.currentText() != self.input.itemData(0, Qt.UserRole):
```

Comment:
Hi @myarjunar I am not sure this is a good way to compare currentText and itemData.
Actually, I suggest to make an attribute in the DefaultSelectParameterWidget class to indicate which string that will trigger the action (hide/show the radio button). So, there will be a set of this attribute in the wizard.